use-cases/eurac/trainer.py

```diff
@@ -0,0 +1,523 @@
+import os
+from copy import deepcopy
+from pathlib import Path
+from timeit import default_timer as timer
+from typing import Dict, Literal, Optional, Union
+
+import pandas as pd
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from hython.losses import RMSELoss
+from hython.metrics import MSEMetric
+from hython.sampler import SamplerBuilder
+from hython.trainer import ConvTrainer, RNNTrainer, RNNTrainParams
+from ray import train
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from tqdm.auto import tqdm
+
+from itwinai.loggers import EpochTimeTracker, Logger
+from itwinai.torch.config import TrainingConfiguration
+from itwinai.torch.distributed import (
+    DeepSpeedStrategy,
+    HorovodStrategy,
+    NonDistributedStrategy,
+    TorchDDPStrategy,
+)
+from itwinai.torch.trainer import TorchTrainer
+from itwinai.torch.type import Metric
+
+
+class RNNDistributedTrainer(TorchTrainer):
+    """Trainer class for RNN model using pytorch.
+
+    Args:
+        config (Union[Dict, TrainingConfiguration]): training configuration
+            containing hyperparameters.
+        epochs (int): number of training epochs.
+        model (Optional[nn.Module], optional): model to train.
+            Defaults to None.
+        strategy (Literal['ddp', 'deepspeed', 'horovod'], optional):
+            distributed strategy. Defaults to 'ddp'.
+        validation_every (Optional[int], optional): run a validation epoch
+            every ``validation_every`` epochs. Disabled if None. Defaults to 1.
+        test_every (Optional[int], optional): run a test epoch
+            every ``test_every`` epochs. Disabled if None. Defaults to None.
+        random_seed (Optional[int], optional): set random seed for
+            reproducibility. If None, the seed is not set. Defaults to None.
+        logger (Optional[Logger], optional): logger for ML tracking.
+            Defaults to None.
+        metrics (Optional[Dict[str, Metric]], optional): map of torch metrics
+            metrics. Defaults to None.
+        checkpoints_location (str): path to checkpoints directory.
+            Defaults to "checkpoints".
+        checkpoint_every (Optional[int]): save a checkpoint every
+            ``checkpoint_every`` epochs. Disabled if None. Defaults to None.
+        name (Optional[str], optional): trainer custom name. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        config: Union[Dict, TrainingConfiguration],
+        epochs: int,
+        model: Optional[nn.Module] = None,
+        strategy: Optional[Literal["ddp", "deepspeed", "horovod"]] = "ddp",
+        validation_every: Optional[int] = 1,
```

Comment:
Same as above with the Optional

Reply:
I don't know about this one as I'm not so versed in the eurac code, but this is at least consistent with how it is in the TorchTrainer

Reply:
This one can also be None, so it is fine the way it is :)